substrate/client/db/src/lib.rs

```diff
@@ -214,6 +214,21 @@ impl<B: BlockT> StateBackend<HashingFor<B>> for RefTrackingState<B> {
 		self.state.child_storage_hash(child_info, key)
 	}
 
+	fn closest_merkle_value(
+		&self,
+		key: &[u8],
+	) -> Result<Option<MerkleValue<B::Hash>>, Self::Error> {
+		self.state.closest_merkle_value(key)
+	}
```

Comment:
Question: What does it mean here, if an Error is returned, vs. an Option is returned? Under what circumstances can errors happen?

Reply:
In this case, the `Option` tells us if there is descendant merkle value below the provided key, since the key may not be in the database and may not have any descendants.

On the other hand, the error is a way to propagate that something went wrong during this operation. Mainly on the trie-db side:
- unable to fetch bytes from the database (invalid state root / incomplete data)
- corrupted data (failing to decode the node)

[Here](https://github.com/paritytech/trie/blob/4d05c641a314d136e1606159856d863094954cdd/trie-db/src/lookup.rs#L154) would be the main place from which those errors originate